### PR TITLE
036 New JSON Implementation + Tests

### LINKED ISSUE #508: Specify JSON validation in ByteStreamLineAtomizer
Currently only '{' are checked to validate a json. This can lead to long checks.
'[' can also be validated to check if an invalid json occurred.

In both cases '{' and '[' can not be distinguished when used in json strings starting with ".

A json parser should be implemented.

Also change the runAminerJsonInputDemo to validate the output of the aminer.

### LINKED ISSUE #509: ByteStreamLineAtomizer json_format use max_line_length
By default the parameter should be None, so no limit is applied.

### LINKED ISSUE #510: ["apple", "bananas"] is valid json data. However we cannot distinguish it from python data.
Every json data must start with { (even lists). Is there another solution?

### PR DESCRIPTION
**Pull request by dITySoftware**
# Make sure these boxes are signed before submitting your Pull Request -- thank you.

# Must haves
- [x] I have read and followed the contributing guide lines at https://github.com/ait-aecid/logdata-anomaly-miner/wiki/Git-development-workflow
- [x] Issues exist for this PR
- [x] I added related issues using the "Fixes #123'-notations
- [x] This Pull-Requests merges into the "development"-branch

Fixes #508
Fixes #510 
Fixes #509 

# Submission specific

- [ ] This PR introduces breaking changes
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

# Describe changes:
-

### CODE REVIEW

File: aecid-testsuite/unit/input/ByteStreamLineAtomizerTest.py [hunk 1]
```diff
@@ -90,6 +90,15 @@ def test6json_format(self):
         json_data = b'{"a": 1, "b": {"c": 2}, "d": 3}\n{"a": 1, "b": {"c": 2}, d'
         self.assertEqual(byte_stream_line_atomizer.consume_data(json_data, False), json_data.rfind(b'\n') + 1)
 
+    def test7json_max_line_length(self):
```
Comment (landauermax) [inline aecid-testsuite/unit/input/ByteStreamLineAtomizerTest.py:93]: Shouldnt the aminer issue an overlong line anomaly? The tests should check whether this anomaly occurs correctly.

Comment (dITySoftware) [reply aecid-testsuite/unit/input/ByteStreamLineAtomizerTest.py:93]: Yes it should. I have revised the 7th test.

Comment (landauermax) [reply aecid-testsuite/unit/input/ByteStreamLineAtomizerTest.py:93]: Ok, thanks.

**Review by landauermax — APPROVED** [review]
Looks good to me

**Pull request merged**